File: src/api/tables/professor.py
from sqlalchemy import Column
from sqlalchemy.dialects.postgresql import VARCHAR

from .database import Base


class Professor(Base):
    __tablename__ = "professor"

    email = Column(VARCHAR(length=255), primary_key=True)
    name = Column(VARCHAR(length=255), nullable=True)
    title = Column(VARCHAR(length=255), nullable=True)
    phone_number = Column(VARCHAR(length=255), nullable=True)
    department = Column(VARCHAR(length=255), nullable=True)
    portfolio_page = Column(VARCHAR(length=255), nullable=True)
    profile_page = Column(VARCHAR(length=255), nullable=True)

    # rcs = Column(VARCHAR(length=255), nullable = True)

File: src/web/src/api/professor.ts
import { useMemoize } from "@vueuse/core";
import { app } from "./common";

export interface Professor {
  name: string;
  title: string;
  email: string;
  phone_number: string;
  department: string;
  portfolio_page: string;
  profile_page: string | null;
};

/** Gets professor info */
export const getProfessors = useMemoize(
  async (): Promise<Professor[]> => {
    const resp = await app.get<Professor[]>("/professor");
    return resp.data;
  }
);
